### DISCUSSION TITLE
Remove Duplicates from an array

### DISCUSSION THREAD

**Discussion by cvsudheer108** [schibsted/jslt#231] [General]
I apologize being a beginner on JSLT, spent quite bit of time to get this simple thing done. Please help.

//============================JSLT===================================

 let paymentTypeIndexMap = {
    "1": 1,
    "4": 15,
    "5": 3,
    "9" : 1,
    "11": 2
  }

let a = array([1,4,9,11])
let b = [for($a) get-key($paymentTypeIndexMap, string(.),"")]

{
"Array" : $b,
"ArrayWithNoDuplicates" : ""
}

**Comment by larsga** [comment]
You can make an object in order to have JSLT deduplicate the keys for you, then turn that back into an array. I agree it's a bit convoluted, but at least it works.

```
{
"Array" : $b,
"ArrayWithNoDuplicates" : [for ({for ($b) string(.) : string(.)}) number(.key)]
}
```

**Reply by cvsudheer108** [reply]
Thanks a lot @larsga!!
